### PR TITLE
fix get_base_name

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/util/get_base_name.cpp [hunk 1]
```diff
@@ -8,31 +8,39 @@ Author: CM Wintersteiger
...
-    f=0;
+  std::size_t start_pos;
 
-  size_t fw=in.rfind('\\', in.length()-1);
-  if(fw==std::string::npos)
-    fw=0;
+  if(slash_pos == std::string::npos)
+    start_pos = 0;
+  else
+    start_pos = slash_pos + 1;
```
Comment (tautschnig) [inline src/util/get_base_name.cpp]: Nit pick: I'd use `?:` here: `const std::size_t start_pos = slash_pos == std::string::npos ? 0 : slash_pos + 1;`

File: src/util/get_base_name.cpp [hunk 2]
```diff
@@ -8,31 +8,39 @@ Author: CM Wintersteiger
...
+  if(strip_suffix)
+  {
+    std::size_t dot_pos = in.rfind('.');
+    if(dot_pos == std::string::npos || dot_pos < start_pos)
+      char_count = std::string::npos;
+    else
+      char_count = dot_pos - start_pos;
+  }
+  else
+    char_count = std::string::npos;
```
Comment (tautschnig) [inline src/util/get_base_name.cpp]: I think a few lines of code can be saved by initialising `char_count` to `std::string::npos` and only updating if `(dot_pos != std::string::npos && dot_pos >= start_pos)`

**Review by tautschnig — APPROVED** [review]
Nit picks below - thanks for the test!!

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 8b51fafb7ce9ffc0d58bf6ba99021e54a78350af).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80883871

**Pull request merged**